src/lib/sql/sql_pipeline_statement.cpp

```diff
@@ -92,6 +92,8 @@ const std::shared_ptr<AbstractLQPNode>& SQLPipelineStatement::get_unoptimized_lo
   DebugAssert(lqp_roots.size() == 1, "LQP translation returned no or more than one LQP root for a single statement.");
   _unoptimized_logical_plan = lqp_roots.front();
 
+  _cacheable = sql_translator.cacheable();
```

Comment:
In any case, we should have some documentation here on why some things are not cacheable.